### PR TITLE
[2022.2] Fixing various API validation issues

### PR DESCRIPTION
**Pull request by ellioman**
# Purpose of this PR
There are quite a few API validation issues currently in URP.
This PR was made to fix those.

Most of the breaking changes from from #6589 and #6616.

# Testing status
Describe what manual/automated tests were performed for this PR

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Debug/DebugDisplaySettingsRendering.cs [hunk 1]
```diff
@@ -42,6 +42,23 @@ public DebugWireframeMode wireframeMode
             }
         }
 
+        bool m_Overdraw = false;
+
+        /// <summary>
+        /// Whether debug overdraw mode is active.
+        /// </summary>
+        [Obsolete("overdraw has been deprecated. Use overdrawMode instead.", false)]
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Debug/DebugDisplaySettingsRendering.cs:50]: When was this obsolete?

File: com.unity.render-pipelines.universal/Editor/Deprecated.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace UnityEditor.Rendering.Universal
     /// Editor script for a <c>ForwardRendererData</c> class.
     /// </summary>
     [Obsolete("ForwardRendererDataEditor has been deprecated. Use UniversalRendererDataEditor instead (UnityUpgradable) -> UniversalRendererDataEditor", true)]
-    internal class ForwardRendererDataEditor : ScriptableRendererDataEditor
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/Deprecated.cs:9]: When was this obsolete?

File: com.unity.render-pipelines.universal/Editor/PostProcessDataEditor.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@ namespace UnityEditor.Rendering.Universal
     /// Editor script for a <c>PostProcessData</c> class.
     /// </summary>
     [CustomEditor(typeof(PostProcessData), true)]
-    internal class PostProcessDataEditor : Editor
+    public class PostProcessDataEditor : Editor
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/PostProcessDataEditor.cs:11]: Note for myself, validate this one as this is editor inspector only.

File: com.unity.render-pipelines.universal/Editor/UniversalAdditionalLightDataEditor.cs [hunk 4]
```diff
@@ -8,7 +8,7 @@ namespace UnityEditor.Rendering.Universal
     /// </summary>
     [CanEditMultipleObjects]
     [CustomEditor(typeof(UniversalAdditionalLightData))]
-    internal class UniversalAdditionLightDataEditor : Editor
+    public class UniversalAdditionLightDataEditor : Editor
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/UniversalAdditionalLightDataEditor.cs:11]: note for myself, validate this one as this one is custom editor

File: com.unity.render-pipelines.universal/Runtime/Deprecated.cs [hunk 5]
```diff
@@ -172,6 +172,7 @@ public sealed partial class Bloom : VolumeComponent, IPostProcessComponent
     {
         // Deprecated in 13.x.x
         [Obsolete("This is obsolete, please use maxIterations instead.", false)]
+        [Tooltip("The number of final iterations to skip in the effect processing sequence.")]
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Deprecated.cs:175]: Deprecated in 13.x.x, seems like we could remove this one already?

File: com.unity.render-pipelines.universal/Runtime/Passes/DepthNormalOnlyPass.cs [hunk 6]
```diff
@@ -32,7 +32,7 @@ public DepthNormalOnlyPass(RenderPassEvent evt, RenderQueueRange renderQueueRang
             useNativeRenderPass = false;
         }
 
-        internal static GraphicsFormat GetGraphicsFormat()
+        public static GraphicsFormat GetGraphicsFormat()
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/DepthNormalOnlyPass.cs:35]: note for myself, discuss this one as this is internal namespace.

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 7]
```diff
@@ -261,12 +261,12 @@ public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderin
...
         }
 
-        private void ResetHistory()
+        public void ResetHistory()
         {
             m_ResetHistory = true;
         }
 
-        private bool CanRunOnTile()
+        public bool CanRunOnTile()
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:269]: Also internal namespace.

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/Debug/DebugDisplaySettingsRendering.cs (hunk 1 above)
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/Debug/DebugDisplaySettingsRendering.cs:50]: I added it back. It was removed here #5883 
If I can see this correctly.

File: com.unity.render-pipelines.universal/Editor/Deprecated.cs (hunk 2 above)
Comment (martint-unity) [reply com.unity.render-pipelines.universal/Editor/Deprecated.cs:9]: Landed here https://github.com/Unity-Technologies/Graphics/pull/1933
As far as I can tell. 12 months ago

**Review by fredericv-unity3d — APPROVED** [review]

**Pull request merged**